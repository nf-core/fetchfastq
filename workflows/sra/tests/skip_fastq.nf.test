nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "sra_skip_fastq"
    tag "workflow"

    test("SRA Skip FASTQ") {

        when {
            params {
                outdir              = "$outputDir"
                skip_fastq_download = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(NftestUtils.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/metadata/DRR026872.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/DRR028935.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/ERR1109373.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/ERR1160846.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/GSE214215.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR12848126.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR13191702.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR14593545.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR14709033.runinfo_ftp.tsv"),
                                path("$outputDir/metadata/SRR9984183.runinfo_ftp.tsv")).match("metadata") },
                { assert new File("$outputDir/metadata/GSM4907283.runinfo_ftp.tsv").exists() },
                { assert snapshot(path("$outputDir/samplesheet/id_mappings.csv"),
                                path("$outputDir/samplesheet/multiqc_config.yml")).match("samplesheet") },
                { assert new File("$outputDir/samplesheet/samplesheet.csv").exists() },
                { assert !new File("$outputDir/fastq/DRX024467_DRR026872.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/DRX026011_DRR028935_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/DRX026011_DRR028935_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/ERX1188904_ERR1109373_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/ERX1188904_ERR1109373_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/ERX1234253_ERR1160846.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX10940790_SRR14593545_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX10940790_SRR14593545_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX11047067_SRR14709033.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX6725035_SRR9984183.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9315476_SRR12848126_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9315476_SRR12848126_2.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9626017_SRR13191702_1.fastq.gz").exists() },
                { assert !new File("$outputDir/fastq/SRX9626017_SRR13191702_2.fastq.gz").exists() }
            )
        }

    }

}
