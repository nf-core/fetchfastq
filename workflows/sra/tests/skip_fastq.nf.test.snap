{
    "metadata": {
        "content": [
            "DRR026872.runinfo_ftp.tsv:md5,c5ab92f6f50717b8163580b642b37aec",
            "DRR028935.runinfo_ftp.tsv:md5,9db5d6cba5ce2cccd9b2cc3faaa092e2",
            "ERR1109373.runinfo_ftp.tsv:md5,d1717cb1638950b7af22ec1b976616a4",
            "ERR1160846.runinfo_ftp.tsv:md5,5611622d583e6bc6d45d8f934696ff1e",
            "GSE214215.runinfo_ftp.tsv:md5,4224b5733b92daa40689538de9498ab9",
            "SRR12848126.runinfo_ftp.tsv:md5,9563f91d4cc6aa58d2ed5fba2eb30baf",
            "SRR13191702.runinfo_ftp.tsv:md5,94378c448c044b3e20e5c54e442ab62e",
            "SRR14593545.runinfo_ftp.tsv:md5,302271a4a1d99865362d0228e6439580",
            "SRR14709033.runinfo_ftp.tsv:md5,e059c4a0dfdddd70c1998a1ea9c1e65f",
            "SRR9984183.runinfo_ftp.tsv:md5,1e299191d23e0f3bd0aabf98c9d8133e"
        ],
        "timestamp": "2023-05-28T15:07:52+0000"
    },
    "samplesheet": {
        "content": [
            "id_mappings.csv:md5,07262807636ce4b50102308eabdcf253",
            "multiqc_config.yml:md5,a24e0e280fa03b95bb678c17e4c36d93"
        ],
        "timestamp": "2023-05-28T15:07:52+0000"
    },
    "software_versions": {
        "content": [
            "{CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, MULTIQC_MAPPINGS_CONFIG={python=3.9.5}, SRA_IDS_TO_RUNINFO={python=3.9.5}, SRA_MERGE_SAMPLESHEET={sed=4.7}, SRA_RUNINFO_TO_FTP={python=3.9.5}, Workflow={nf-core/fetchngs=1.10.1}}"
        ],
        "timestamp": "2023-05-28T15:07:52+0000"
    }
}